eng/native/functions.cmake

```diff
@@ -367,26 +367,54 @@ function(generate_exports_file_prefix inputFilename outputFilename prefix)
-    else ()
-      set(strip_destination_file $<TARGET_FILE:${targetName}>.dbg)
-    endif ()
-
-    set(${outputSymbolFilename} ${strip_destination_file} PARENT_SCOPE)
-  elseif(CLR_CMAKE_HOST_WIN32)
-    # We can't use the $<TARGET_PDB_FILE> generator expression here since
-    # the generator expression isn't supported on resource DLLs.
-    set(${outputSymbolFilename} $<TARGET_FILE_DIR:${targetName}>/$<TARGET_FILE_PREFIX:${targetName}>$<TARGET_FILE_BASE_NAME:${targetName}>.pdb PARENT_SCOPE)
+  get_target_property(targetImportedNativeAotLib "${targetName}" CLR_IMPORTED_NATIVEAOT_LIBRARY)
+  if (NOT "${targetImportedNativeAotLib}")
+    if (CLR_CMAKE_HOST_UNIX)
+      if (CLR_CMAKE_TARGET_APPLE)
+        set(strip_destination_file $<TARGET_FILE:${targetName}>.dwarf)
+      else ()
+        set(strip_destination_file $<TARGET_FILE:${targetName}>.dbg)
+      endif ()
+
+      set(${outputSymbolFilename} ${strip_destination_file} PARENT_SCOPE)
+    elseif(CLR_CMAKE_HOST_WIN32)
+      # We can't use the $<TARGET_PDB_FILE> generator expression here since
+      # the generator expression isn't supported on resource DLLs.
+      set(${outputSymbolFilename} $<TARGET_FILE_DIR:${targetName}>/$<TARGET_FILE_PREFIX:${targetName}>$<TARGET_FILE_BASE_NAME:${targetName}>.pdb PARENT_SCOPE)
+    endif()
+  else()
+    get_property(libraryType TARGET ${targetName} PROPERTY TYPE)
+    message(TRACE "Target ${targetName} is imported of type ${libraryType}")
+    if ("${libraryType}" STREQUAL "SHARED_LIBRARY")
+      get_property(importedLocation TARGET ${targetName} PROPERTY IMPORTED_LOCATION)
+      message(TRACE "Target ${targetName} is imported from  ${importedLocation}")
+      if (CLR_CMAKE_HOST_UNIX)
+        if (CLR_CMAKE_TARGET_APPLE)
+          set(importedLocation "${importedLocation}.dwarf")
+        else()
+          set(importedLocation "${importedLocation}.dbg")
+        endif()
+      elseif(CLR_CMAKE_HOST_WIN32)
+        cmake_path(REPLACE_EXTENSION importedLocation ".pdb")
+      endif()
+      set(${outputSymbolFilename} "${importedLocation}" PARENT_SCOPE)
+      message(TRACE "Target ${targetName} symbols will be in ${importedLocation}")
+    endif()
   endif()
 endfunction()
 
 function(strip_symbols targetName outputFilename)
   get_symbol_file_name(${targetName} strip_destination_file)
   set(${outputFilename} ${strip_destination_file} PARENT_SCOPE)
   if (CLR_CMAKE_HOST_UNIX)
-    set(strip_source_file $<TARGET_FILE:${targetName}>)
+    get_target_property(targetImportedNativeAotLib "${targetName}" CLR_IMPORTED_NATIVEAOT_LIBRARY)
+    if (NOT "${targetImportedNativeAotLib}")
+      set(strip_source_file $<TARGET_FILE:${targetName}>)
+    else()
+      get_property(strip_source_file TARGET ${targetName} PROPERTY IMPORTED_LOCATION)
+      # for an imported library, we will hang the post-build event on the copy target
+      get_property(copy_target TARGET ${targetName} PROPERTY CLR_IMPORTED_NATIVEAOT_LIBRARY_COPY_TARGET)
```

Comment:
Could we rename `CLR_IMPORTED_NATIVEAOT_LIBRARY_COPY_TARGET` to `CLR_IMPORTED_COPY_TARGET` and unconditionally check for it instead of conditioning it on `CLR_IMPORTED_NATIVEAOT_LIBRARY`?
And for the other cases, could we use the `IMPORTED` target property?

Reply:
> And for the other cases, could we use the `IMPORTED` target property?

I didn't want to prescribe behavior for other imported libraries

> Could we rename CLR_IMPORTED_NATIVEAOT_LIBRARY_COPY_TARGET to CLR_IMPORTED_COPY_TARGET and unconditionally check for it instead of conditioning it on CLR_IMPORTED_NATIVEAOT_LIBRARY?

yes

Reply:
On second thought, I ended up making a reusable `add_imported_library_clr` function that just does the copy/strip/dependency handling and reimplemented `add_imported_nativeaot_library` on top of that.  `CLR_IMPORTED_COPY_TARGET` is now the property that the stripping logic looks for, and `CLR_IMPORTED_NATIVEAOT_LIBRARY` is just a marker for NativeAOT libs that we don't currently use for anything.